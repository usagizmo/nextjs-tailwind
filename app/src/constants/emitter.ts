export const EMITTER = {
  scrollTop: 'SCROLL_TOP',
  animatedScrollTop: 'ANIMATED_SCROLL_TOP',
  bodyHeight: 'BODY_HEIGHT',
  bodyOverflowY: 'BODY_OVERFLOW_Y',
  mouseMove: 'MOUSE_MOVE',
}
