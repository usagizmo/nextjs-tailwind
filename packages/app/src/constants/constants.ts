export const TWITTER_NAME = 'usagizmo'
