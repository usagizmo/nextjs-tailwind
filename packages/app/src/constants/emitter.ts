export const EMITTER = {
  animatedScrollTop: 'ANIMATED_SCROLL_TOP',
  animationFrame: 'ANIMATION_FRAME',
  bodyHeight: 'BODY_HEIGHT',
  mouseDown: 'MOUSE_DOWN',
  mouseMove: 'MOUSE_MOVE',
  mouseUp: 'MOUSE_UP',
  scrollTop: 'SCROLL_TOP',
}
