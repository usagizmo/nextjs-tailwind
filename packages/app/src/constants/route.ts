export const ROUTE = {
  HOME: '/',
  DETAIL_ID: '/detail/[id]',
}
