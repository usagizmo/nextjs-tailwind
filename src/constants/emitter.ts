export const EMITTER = {
  scrollY: 'SCROLL_Y',
  animatedScrollY: 'ANIMATED_SCROLL_Y',
  windowSize: 'WINDOW_SIZE',
}
