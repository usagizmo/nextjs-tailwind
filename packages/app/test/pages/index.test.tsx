import React from 'react'
import { mockRouter, render } from '../test-utils'
import Home from '../../src/pages/index'
import { ROUTE } from '../../src/constants/route'

describe('Home page', () => {
  it('matches snapshot', () => {
    const router = {
      ...mockRouter,
      pathname: ROUTE.HOME,
    }
    const { asFragment } = render(<Home />, { router })
    expect(asFragment()).toMatchSnapshot()
  })

  // it('clicking button triggers alert', () => {
  //   const { getByText } = render(<Home />)
  //   window.alert = jest.fn()
  //   fireEvent.click(getByText('Test Button'))
  //   expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  // })
})
