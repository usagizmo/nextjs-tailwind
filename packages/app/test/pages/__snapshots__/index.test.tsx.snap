// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="css-3x0lj0-size-center-column-IndexPage"
  >
    <p
      class="css-1i7eqxm-IndexPage"
    >
      HOME
    </p>
  </main>
</DocumentFragment>
`;
